src/scripts/main.js

```diff
@@ -1,5 +1,34 @@
 'use strict';
 
-document.addEventListener('click', (e) => {
-  // write code here
+const wall = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+
+// Убедитесь, что ширина границы учитывается, если это необходимо
+const borderSize = parseInt(getComputedStyle(wall).borderWidth) || 0;
+
+wall.addEventListener('click', (e) => {
+  const wallRect = wall.getBoundingClientRect();
+  const spiderWidth = spider.offsetWidth;
+  const spiderHeight = spider.offsetHeight;
+
+  const offsetX = e.clientX - wallRect.left;
+  const offsetY = e.clientY - wallRect.top;
+
+  let newX = offsetX - spiderWidth / 2 - borderSize;
+  let newY = offsetY - spiderHeight / 2 - borderSize;
```

Comment:
The subtraction of `borderSize` in both `newX` and `newY` calculations is incorrect. Since `borderSize` is already accounted for in the boundary checks (lines 23 and 28), it should not be subtracted here. Remove `- borderSize` from these calculations to fix the issue.